torch/lib/c10d/test/ProcessGroupGlooTest.cpp

```diff
@@ -268,87 +269,67 @@ void testSend(const std::string& path) {
   auto tests = CollectiveTest::initialize(path, size);
 
   constexpr uint64_t tag = 0x1337;
-
-  // Create a sender thread and ensure that sendWork can be aborted.
-  std::thread senderThread([&]() {
-    auto selfRank = 0;
-    auto destRank = 1;
-
-    auto& pg = tests[selfRank].getProcessGroup();
-
-    std::vector<at::Tensor> tensors = {
-        at::ones({16, 16}),
-    };
-    // Ensure that we can abort SendWork
-    auto work = pg.send(tensors, destRank, tag);
-    bool sendCompleted;
-    std::thread waitSendWorkThread([&]() { sendCompleted = work->wait(); });
-    work->abort();
-    waitSendWorkThread.join();
-    TORCH_CHECK(!sendCompleted);
-
-    // Ensure that future waits can complete successfully.
-    work = pg.send(tensors, destRank, tag);
-    sendCompleted = work->wait();
-    TORCH_CHECK(sendCompleted);
-  });
-
-  // Helper receiver thread to simulate a real send/recv pair.
-  std::thread receiverThread([&]() {
+  // test that waiting for work to be sent can be aborted successfully.
```

Comment:
nit: much of the time, abort() may be called prior to wait() below, depending on OS/etc

If you want, there are ways to increase the probability of wait() being called before abort(), e.g. the equivalent of folly::Baton<> to wait for the thread body to be reached prior to calling abort()

Reply:
In terms of correctness, this shouldn't be an issue, as even if we call `abort()` before `wait()` the wait should still be successfully interrupted. Although if we want to try to test the wait() before abort() sequence, then we could go with a baton-based approach.